### PR TITLE
Fix GCInfo library used by ARM/ARM64 protononjit builds

### PR DESCRIPTION
**Pull request by BruceForstall**
Otherwise, using the normal GCInfo library, the altjits will
assert creating GC info.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@briansull @CarolEidt PTAL
cc @dotnet/jit-contrib

File: src/gcinfo/CMakeLists.txt [hunk 1]
```diff
@@ -23,5 +23,15 @@ endif(CLR_CMAKE_PLATFORM_UNIX)
...
+if (CLR_CMAKE_PLATFORM_ARCH_I386)
+    # On x86, build RyuJIT/ARM32 cross-compiling altjit.
+    add_subdirectory(arm_altjit)
+endif ()
+
+if (CLR_CMAKE_PLATFORM_ARCH_AMD64)
+    # On amd64, build RyuJIT/ARM64 cross-compiling altjit.
+    add_subdirectory(arm64_altjit)
+endif ()
+
```
Comment (briansull) [inline src/gcinfo/CMakeLists.txt]: Does this work for Desktop builds, or is it just for CoreCLR builds?

Comment (BruceForstall) [inline src/gcinfo/CMakeLists.txt]: We already have this logic in the desktop build system; this change is to add it to the cmake-based CoreCLR build system.

File: src/gcinfo/CMakeLists.txt [hunk 2]
```diff
@@ -23,5 +23,15 @@ endif(CLR_CMAKE_PLATFORM_UNIX)
 add_subdirectory(lib)
 add_subdirectory(crossgen)
 
+if (CLR_CMAKE_PLATFORM_ARCH_I386)
+    # On x86, build RyuJIT/ARM32 cross-compiling altjit.
+    add_subdirectory(arm_altjit)
```
Comment (pgavlin) [inline src/gcinfo/CMakeLists.txt]: I don't love that we have `altjit` in the paths here, but I don't have a great alternative. `arm{,64}_cross_gcinfo`, maybe?

Comment (pgavlin) [inline src/gcinfo/CMakeLists.txt]: Or just match the directories to the names of the libraries (e.g. `add_subdirectory(gcinfo_arm64)`)

Comment (BruceForstall) [inline src/gcinfo/CMakeLists.txt]: Why don't you like "altjit" in the path?

Comment (pgavlin) [inline src/gcinfo/CMakeLists.txt]: Because these are builds of the `gcinfo` library, and the `gcinfo` library is logically distinct from the JIT.

Comment (BruceForstall) [inline src/gcinfo/CMakeLists.txt]: But these builds are only used by altjit.

Comment (BruceForstall) [inline src/gcinfo/CMakeLists.txt]: (and gcinfo is only used by JIT, for that matter)

Comment (pgavlin) [inline src/gcinfo/CMakeLists.txt]: Regardless, this is a separate component and should be named as such. It is confusing to navigate to a directory named "gcinfo" and find subdirectories referencing the JIT. For example, when I first looked at the PR, I thought that you had actually placed the directories underneath the JIT's directory (e.g. `src/jit/arm{,64}_altjit`).

**Comment by BruceForstall** [comment]
Updated

**Pull request merged**